### ISSUE TITLE
Non-idempotent jobs do not ever show as exit on a Redis broker

### ISSUE THREAD

**Issue by 0xDEC0DE** [NicolasLM/spinach#45]
# Steps to reproduce

## Prep
In one window:
```
docker-compose -f spinach/tests/docker-compose.yml up -d
docker-compose -f spinach/tests/docker-compose.yml exec redis redis-cli
```

## Test
In another window:
```
import spinach
sp = spinach.Engine(spinach.RedisBroker())

@sp.task(name='hi_mom')
def hi_mom():
    print("Hi mom.")

sp.schedule('hi_mom')
sp.start_workers(stop_when_queue_empty=True)
```

## Check
In the first window:
```
KEYS *
```

# Expected result

An empty result, or an idempotency key.

# Actual behavior

A `_running-jobs-on-broker-*` key is listed in the output.  Its contents are the job that just finished, claiming to still be running.

**Comment by juledwar** [comment]
Can you close this @0xDEC0DE , I don't have the powers.

**Comment by 0xDEC0DE** [comment]
Fixed in #46
